packages/toolpad-app/src/toolpad/Apps/index.tsx

```diff
@@ -138,7 +140,7 @@ function CreateAppDialog({ onClose, ...props }: CreateAppDialogProps) {
     null,
   );
 
-  const isFormValid = config.isDemo || Boolean(name);
+  const isFormValid = Boolean(name);
```

Comment:
Not sure if this is was accidental during resolving conflicts, but we can assume that the form is valid in demo mode because we don't have a name field there anymore (we have automatic names)